src/standalone/Snow/snow_parameterizations.jl

```diff
@@ -372,3 +376,178 @@ function energy_from_T_and_swe(S::FT, T::FT, parameters) where {FT}
+    newsnow_density(air_temp::FT)::FT where {FT}
+Estimates the density of newly fallen snow as a function of air temperature, according to the Snow17 model.
+Used in computing the time derivative of snow depth under the Anderson1976 density parameterization.
+"""
+function newsnow_density(air_temp::FT)::FT where {FT}
+    if air_temp < -15
+        return FT(0.05)
+    else
+        return FT(0.05) + FT(0.0017) * FT((air_temp + FT(15.0))^1.5)
+    end
+end
+
+"""
+    newsnow_temp(air_temp::FT)::FT where {FT}
+Estimates the temperature of newly fallen snow as a function of air temperature, according to the Snow17 model.
+Used in computing the time derivative of snow depth under the Anderson1976 density parameterization.
+"""
+function newsnow_temp(air_temp::FT)::FT where {FT}
+    return (air_temp > 0) ? FT(0) : air_temp
+end
+
+"""
+    dzdt(density::Anderson1976{FT}, model::SnowModel{FT}, Y, p, t) where {FT}
+Returns the change in snow depth (rate) given the current model state and the `Anderson1976`
+density paramterization, which estimates contributions from new snowfall, as well as compaction
+effects from the existing and newly fallen snow.
+"""
+function dzdt(density::Anderson1976{FT}, model::SnowModel{FT}, Y, p) where {FT}
+    Δt_t = model.parameters.Δt / 3600 #in hours (when Δt in SnowParameters is ::FT instead of ::Period)
+
+    #Contribution from new snowfall: (uses parameterization of newsnow_temp, newsnow_density)
+    air_temp = p.drivers.T .- model.parameters.earth_param_set.T_freeze
+    snowfall = abs.(p.drivers.P_snow) .* model.parameters.Δt
+    T_newsnow = newsnow_temp.(p.drivers.T)
+    ρ_precip = newsnow_density.(air_temp) #unitless
+    ρ_newsnow =
+        compact_density.(
+            snowfall,
+            FT(0),
+            ρ_precip,
+            Δt_t,
+            T_newsnow,
+            Ref(density),
+        )
+    Δz = snowfall ./ ρ_newsnow
+
+    #Estimate resulting change from compaction effects of old snow (and the incoming snow on top):
+    W_ice = (1 .- p.snow.q_l) .* Y.snow.S
+    ρ_ice = W_ice ./ Y.snow.Z #unitless
+    sliq = p.snow.q_l .* Y.snow.S
+    Tsnow = p.snow.T .- model.parameters.earth_param_set.T_freeze
+    W_use = W_ice .+ snowfall
+    parent(ρ_ice)[parent(W_ice) .<= eps(FT)] .= FT(0.1) #any nonzero value for no-snowpack to avoid NaN and return 0.0
+    ρ_est = compact_density.(W_use, sliq, ρ_ice, Δt_t, Tsnow, Ref(density))
+    dz_compaction = (W_ice ./ ρ_est) .- Y.snow.Z
+
+    return (dz_compaction .+ Δz) ./ model.parameters.Δt
+end
+
+"""
+    clip_dZdt(S::FT, Z::FT, dSdt::FT, dZdt::FT, Δt::FT)::FT
+A helper function which clips the tendency of Z such that
+its behavior is consistent with that of S.
+"""
+function clip_dZdt(S::FT, Z::FT, dSdt::FT, dZdt::FT, Δt::FT)::FT where {FT}
+    if (S + dSdt * Δt) <= eps(FT)
+        return -Z / Δt
+    elseif (Z + dZdt * Δt) < (S + dSdt * Δt)
+        #do we need to do something about setting q_l = 1 here?
+        return (S - Z) / Δt + dSdt
```

Comment:
Could you describe where this comes from?

Reply:
I was mixing up computational order on auxillary vs prognostic variables and which values would get reported with the value at the time of saving, i've removed this now. (aka, if the snowpack is melted to z=SWE=0, in the saved values would we see z=SWE=0 and q_l = something < 1? happy to talk about this).